### PR TITLE
fix(warnings): fix some warnings about the stack protector not working

### PR DESCRIPTION
**Pull request by sudden6**
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/4007)

### CODE REVIEW

File: src/core/core.cpp [hunk 1]
```diff
@@ -870,11 +870,14 @@ QPair<QByteArray, QByteArray> Core::getKeypair() const
     if (!tox)
         return keypair;
 
-    char buf[std::max(TOX_PUBLIC_KEY_SIZE, TOX_SECRET_KEY_SIZE)];
-    tox_self_get_public_key(tox, (uint8_t*)buf);
-    keypair.first = QByteArray(buf, TOX_PUBLIC_KEY_SIZE);
-    tox_self_get_secret_key(tox, (uint8_t*)buf);
-    keypair.second = QByteArray(buf, TOX_SECRET_KEY_SIZE);
+    QByteArray pk = QByteArray(TOX_PUBLIC_KEY_SIZE, 0x00);
```
Comment (Diadlo) [inline src/core/core.cpp]: Please, use `QByteArray pk(TOX_PUBLIC_KEY_SIZE, 0x00);`

File: src/core/coreencryption.cpp [hunk 2]
```diff
@@ -64,14 +64,14 @@ QByteArray Core::encryptData(const QByteArray &data)
 
 QByteArray Core::encryptData(const QByteArray& data, const Tox_Pass_Key& encryptionKey)
 {
-    uint8_t encrypted[data.size() + TOX_PASS_ENCRYPTION_EXTRA_LENGTH];
+    QByteArray encrypted = QByteArray(data.size() + TOX_PASS_ENCRYPTION_EXTRA_LENGTH, 0x00);
```
Comment (Diadlo) [inline src/core/coreencryption.cpp]: Same as above

File: src/core/coreencryption.cpp [hunk 3]
```diff
@@ -89,18 +89,18 @@ QByteArray Core::decryptData(const QByteArray& data, const Tox_Pass_Key& encrypt
...
     if (data.size() < TOX_PASS_ENCRYPTION_EXTRA_LENGTH)
     {
-        qWarning() << "Not enough data:"<<data.size();
+        qWarning() << "Not enough data:" << data.size();
         return QByteArray();
     }
-    int sz = data.size() - TOX_PASS_ENCRYPTION_EXTRA_LENGTH;
-    uint8_t decrypted[sz];
+    int decryptedSize = data.size() - TOX_PASS_ENCRYPTION_EXTRA_LENGTH;
+    QByteArray decrypted = QByteArray(decryptedSize, 0x00);
```
Comment (Diadlo) [inline src/core/coreencryption.cpp]: Same as above

File: src/core/core.cpp (hunk 1 above)
Comment (sudden6) [reply src/core/core.cpp]: Is there a semantic difference or is it just different syntax?

**Comment by Diadlo** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/core/core.cpp, line 873 at r1](https://reviewable.io:443/reviews/qtox/qtox/4007#-KZqai6roXStT8C5nPYC-r1-873:-KZqv-ll-glFkjvF2jHz:b-y3nqm4) ([raw file](https://github.com/qtox/qtox/blob/65076320ab1a30fe9b3fab8c37713ef7ef0966ee/src/core/core.cpp#L873)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Is there a semantic difference or is it just different syntax?
</blockquote></details>

AFAIK, `=` provide copying

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4007#-:-KZqvBeg9S9Hb6mtHV4t:bnfp4nl)*

**Pull request merged**